torchchat/model.py

```diff
@@ -126,7 +125,10 @@ def forward(
                 dtype=torch.int,
             )
 
-        return self.decoder(decoder_input, input_pos=input_pos)
+            return decoder_input.shape[1], self.decoder(decoder_input, input_pos=input_pos)
+        else:
+            return self.decoder(decoder_input, input_pos=input_pos)
```

Comment:
Not sure I follow why we need to do this

Reply:
This is for returning the context length for prefilling to set up the pos_id for the generation after prefilling.
Context is we need to set the correct pos_id for generation after prefilling, so that we need to know how many tokens have been consumed by text decoder during prefilling stage. However, that's impossible to get the number by only counting the number of text tokens forwarded to model like what we do for text-only models since we also insert the image tokens in the middle, and that is generated by image head during the model inference. 
Therefore here we asked model to return the size of decoder input, and use that to calculate pos_id for following generation,